### PR TITLE
bsim: Generate coverage reports

### PR DESCRIPTION
**Pull request by hermabe**
The script works locally, but I am not so familiar with the github workflows.

Dont know if I can force the codecov job to run on a PR.

### CODE REVIEW

File: .github/workflows/codecov.yaml [hunk 1]
```diff
@@ -110,7 +110,7 @@ jobs:
 
   codecov-results:
     name: "Publish Coverage Results"
-    needs: codecov
+    needs: [codecov, bsim-test]
```
Comment (stephanosio) [inline .github/workflows/codecov.yaml:113]: That is not how `workflow_call` works. Please see https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

**Review by stephanosio — CHANGES_REQUESTED** [review]

**Comment by stephanosio** [comment]
> Dont know if I can force the codecov job to run on a PR.

We do not currently run code coverage on pull requests.

**Comment by Thalley** [comment]
@hermabe Any progress on this?

**Comment by Thalley** [comment]
@aescolar Is this something you would be willing to take a stab at?

**Comment by aescolar** [comment]
I think I would need @stephanosio help. Let's wait until he is back.

**Pull request closed without merging**